mono/utils/mono-publib.h

```diff
@@ -144,6 +144,12 @@ mono_set_allocator_vtable (MonoAllocatorVTable* vtable);
 #pragma GCC diagnostic error "-Wdeprecated-declarations"
 #endif
 
+#elif defined (_MSC_VER)
+#define MONO_RT_EXTERNAL_ONLY __declspec (deprecated ("The mono runtime must not call this function.")) \
```

Comment:
The version that takes a message is "new". Old enough?

Reply:
(just like gcc)

Reply:
To be safe, if _MSC_VER greater than something. To be deliberately harsh with old compilers, not a bad thing, asis.

Reply:
What's our lower bound compiler on Windows? Is it VS2015 or something older?

The oldest MSDN docs that mention deprecated with a message that I can see are for VS2008.

Reply:
Should be fine, in master we have upgraded the solution and vcxproj's to VS2015. In other places (like mkbundle) we have dropped support for VS2013 toolchain as well, so if this was introduced in VS2008 then just using _MCS_VER should be enough.